### ISSUE TITLE
Special Case `this` for es6.spread?

### ISSUE THREAD

**Issue by rwjblue** [babel/babel#1033]
Input:

```javascript
var obj = {
  foo: function foo() {
    this.bar(...arguments)
    this.blah(...arguments)
  }
}
```

Output:

```javascript
"use strict";

var obj = {
  foo: function foo() {
    var _ref, _ref2;

    (_ref = this).bar.apply(_ref, arguments);
    (_ref2 = this).blah.apply(_ref2, arguments);
  }
};
```

Memoizing makes tons of sense, but in the case of `this` I would think that the following output would be sufficient:

```javascript
var obj = {
  foo: function foo() {
    this.bar.apply(this, arguments);
    this.blah.apply(this, arguments);
  }
}
```

What am I missing?

**Comment by sebmck** [comment]
> What am I missing?

Nothing :stuck_out_tongue_winking_eye: Should be a quick fix.

**Comment by eventualbuddha** [comment]
@sebmck I'll take this one, if you haven't already.

**Comment by sebmck** [comment]
@eventualbuddha Already done, didn't push my commit, whoops.

**Comment by eventualbuddha** [comment]
I have a test for it, if you think that'd be useful.

**Comment by sebmck** [comment]
Sure.

**Comment by sebmck** [comment]
Sorry for the delay, released as of 4.7.13. Thanks!
